### ISSUE TITLE
AdGuard encrypted DNS using iOS/macOS config profiles fails to reconnect after using a VPN

### ISSUE THREAD

**Issue by hampton8** [AdguardTeam/AdGuardDNS#125]
When using AdGuardDNS on iPhone, iPad or Mac using the new encrypted DNS configuration profiles (available to download on AdGuard's website [here](https://adguard.com/en/blog/encrypted-dns-ios-14.html)), after connecting and disconnecting to/from a VPN service, the device reverts back to using the default ISP DNS, despite the Settings app still showing AdGuard as selected. This is occurring on iOS 14.2, iPadOS 14.2 and macOS Big Sur 11.0.1 (the latest public versions of all 3), and the only way to get the device to use AdGuardDNS again seems to be to reboot.

Here is how I tested the issue:

1. Install AdGuard DNS profile (also tested with NextDNS)
2. Select it as the DNS provider in settings (restart was required on iPhone for this to take effect, but strangely not on iPad or Mac)
3. Open ProtonVPN and start a connection (also tested with Windscribe) 
4. Terminate VPN
5. Settings app still shows AdGuard DNS as active but DNSLeakTest.com shows my ISP's DNS servers
6. Only fix is to restart the device

There is a Reddit thread discussing this bug on r/AdGuard, available [here](https://www.reddit.com/r/Adguard/comments/k0pufu/dns_problems_on_macos_big_sur_and_ios_14_not_sure/).

Since this affects both AdGuard and NextDNS, it seems this is probably a bug with the OS, and not an issue on AdGuard's part - but I thought I would post it here just to make the team aware that it exists. I have reported the bug to Apple via their Feedback Assistant webpage, and I suggest others do the same.

Strangely, last night, the bug seemed to temporarily disappear since I was able to connect to ProtonVPN on my iPhone and iPad, and then disconnect and go straight back to using AdGuardDNS as expected. However, I then tested this on my Mac and found that the bug was very much still there, and then tried Windscribe VPN on my iPhone and iPad and once again, upon termination of the VPN, my devices reverted to using my ISP's DNS server.

**Comment by ameshkov** [comment]
Yeah, it does sound like one more Apple bug with new DNS profiles. Good to know anyways.

**Comment by hampton8** [comment]
@ameshkov No worries :) Hopefully they can become more stable soon, since they're a far more elegant solution (in my opinion) than a dummy VPN! Will the option to configure DNS using a profile instead of a VPN be coming to the AdGuard iOS app soon?

**Comment by ameshkov** [comment]
> Will the option to configure DNS using a profile instead of a VPN be coming to the AdGuard iOS app soon?

Yeah, it's coming in v4.1. Not sure if we'll be able to publish the release this year, but it'll get to beta testing relatively soon.

**Comment by Eow2022yia** [comment]
> When using AdGuardDNS on iPhone, iPad or Mac using the new encrypted DNS configuration profiles (available to download on AdGuard's website [here](https://adguard.com/en/blog/encrypted-dns-ios-14.html)), after connecting and disconnecting to/from a VPN service, the device reverts back to using the default ISP DNS, despite the Settings app still showing AdGuard as selected. This is occurring on iOS 14.2, iPadOS 14.2 and macOS Big Sur 11.0.1 (the latest public versions of all 3), and the only way to get the device to use AdGuardDNS again seems to be to reboot.
> 
> Here is how I tested the issue:
> 
> 1. Install AdGuard DNS profile (also tested with NextDNS)
> 2. Select it as the DNS provider in settings (restart was required on iPhone for this to take effect, but strangely not on iPad or Mac)
> 3. Open ProtonVPN and start a connection (also tested with Windscribe)
> 4. Terminate VPN
> 5. Settings app still shows AdGuard DNS as active but DNSLeakTest.com shows my ISP's DNS servers
> 6. Only fix is to restart the device
> 
> There is a Reddit thread discussing this bug on r/AdGuard, available [here](https://www.reddit.com/r/Adguard/comments/k0pufu/dns_problems_on_macos_big_sur_and_ios_14_not_sure/).
> 
> Since this affects both AdGuard and NextDNS, it seems this is probably a bug with the OS, and not an issue on AdGuard's part - but I thought I would post it here just to make the team aware that it exists. I have reported the bug to Apple via their Feedback Assistant webpage, and I suggest others do the same.
> 
> Strangely, last night, the bug seemed to temporarily disappear since I was able to connect to ProtonVPN on my iPhone and iPad, and then disconnect and go straight back to using AdGuardDNS as expected. However, I then tested this on my Mac and found that the bug was very much still there, and then tried Windscribe VPN on my iPhone and iPad and once again, upon termination of the VPN, my devices reverted to using my ISP's DNS server.

Hi,

I am aware of this issue, it is indeed a bug, well what it seems like.

However this bug only affects IKEv2 connections (also known as Personal VPN).

This can be fixed by simply deleting the full VPN configuration after use, not just disconnecting.

This issue can be avoided all together by using the WireGuard or OpenVPN protocol as this bug only impacts Personal VPN’s.

Hope this helps, thank you.

**Comment by ottoosworth** [comment]
according to the post here https://discuss.privacyguides.net/t/ios-dns-profile-private-dns-icloud-private-relay-domains-not-blocked/32894 the user mentions the problem being the profiles themselves and being able to solve this, not sure if that has been tried
